tests/sqlparser_duckdb.rs

```diff
@@ -32,6 +32,36 @@ fn duckdb_and_generic() -> TestedDialects {
     }
 }
 
+#[test]
+fn test_struct() {
+    // basic struct
+    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, i INTEGER))"#;
+    let statement = duckdb().verified_stmt(sql);
+    assert_eq!(statement.to_string(), sql);
+
+    // struct array
+    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, i INTEGER)[])"#;
+    let statement = duckdb().verified_stmt(sql);
+    assert_eq!(statement.to_string(), sql);
+
+    // nested struct
+    let sql = r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, s STRUCT(a1 INTEGER, a2 VARCHAR))[])"#;
+    let statement = duckdb().verified_stmt(sql);
+    assert_eq!(statement.to_string(), sql);
+
+    // failing test
+    let sql_list = vec![
+        r#"CREATE TABLE t1 (s STRUCT(v VARCHAR, i INTEGER)))"#,
```

Comment:
we can include a test case that fails if no fieldname is provided? since that's not optional for this dialect

Reply:
added